phpseclib/File/ASN1.php

```diff
@@ -333,6 +333,9 @@ private static function decode_ber($encoded, $start = 0, $encoded_pos = 0)
                 $remainingLength = $length;
                 while ($remainingLength > 0) {
                     $temp = self::decode_ber($content, $start, $content_pos);
+                    if (false == $temp) {
+                        break;
```

Comment:
No where else in the phpseclib are [yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions) used. As such, for consistency purposes, I don't think they ought to be used anywhere. ie. they ought to be used everywhere or nowhere.

Also, every other condition does `===` but this one does `==`.

Finally, I think `return false` would be better than `break`.

That said, don't worry about these changes - I cherry-picked this branch into the 1.0 branch and made the changes there. I then merged to 2.0 and master.

Thanks!

Reply:
Ah .. good catch and thnx for fixing it and the pull.